packages/shared/src/components/squads/SquadMemberRoleBadge.tsx

```diff
@@ -0,0 +1,48 @@
+import classNames from 'classnames';
+import React, { ReactElement, useMemo } from 'react';
+import { Author } from '../../graphql/comments';
+import { Source, SourceMemberRole } from '../../graphql/sources';
+import StarIcon from '../icons/Star';
+import UserIcon from '../icons/User';
+
+export type SquadMemberRoleBadgeProps = {
+  className?: string;
+  source?: Source;
+  author?: Author;
+};
+
+const RoleToIconMap = {
+  [SourceMemberRole.Moderator]: UserIcon,
+  [SourceMemberRole.Owner]: StarIcon,
+};
+
+const SquadMemberRoleBadge = ({
+  className,
+  author,
+  source,
+}: SquadMemberRoleBadgeProps): ReactElement => {
+  const role = useMemo(() => {
+    return source?.privilegedMembers?.find(
+      (member) => member.user.id === author?.id,
+    )?.role;
+  }, [author, source]);
+  const RoleIcon = RoleToIconMap[role];
+
+  if (!RoleIcon) {
+    return null;
+  }
+
+  return (
+    <span
+      className={classNames(
+        'flex items-center ml-2 text-cabbage-40 typo-footnote font-bold capitalize',
+        className,
+      )}
+    >
+      <RoleIcon secondary className="mx-0.5" />
+      {role}
+    </span>
```

Comment:
I think we should instead create a simple component that will accept an icon and its label. I see this kind of component used when the user is the Author or Scout. Maybe it is time to create a single component to be used for badges like this.

Then we will simplify the `CommentAuthor` component to just receive an array of badges we will then map to return the newly introduced component. Technically the comment doesn't really need to know about/get the `source` object.

Reply:
I see where you are coming from, for sure we can generalize badges view component. 

Still I think we would need components like `SquadMemberRoleBadge` as business logic wrappers to contain logic for extracting role since it would be clunky to have that logic spread into all of the components where badges are used. 

I'll look into this today.

Reply:
Totally agree with that. I always love seeing hooks getting introduced. We can probably use that to wrap the logic outside this component as you mentioned. Way to go! 🚢